dbms/src/Storages/System/StorageSystemParts.cpp

```diff
@@ -102,4 +107,76 @@ void StorageSystemParts::processNextStorage(MutableColumns & columns, const Stor
     }
 }
 
+/**
+  * Implements system table 'detached_parts' which allows to get information
+  * about detached data parts for tables of MergeTree family.
+  * We don't use StorageSystemPartsBase, because it introduces virtual _state
+  * column and column aliases which we don't need.
+  */
+class StorageSystemDetachedParts :
```

Comment:
Better to move to separate file (one storage one file principle).

Reply:
(it will be even more justified when we will move documentation elements to code)